app/app/schemas/year.py

```diff
@@ -6,11 +6,18 @@
 
 # Shared properties
 class YearBase(BaseModel):
+    name: Optional[str] = None
     school_id: Optional[int] = None
     start_year: Optional[int] = None
     end_year: Optional[int] = None
     is_active: Optional[bool] = True
 
+    @validator('name')
+    def name_not_empty(cls, name: str) -> str:
+        if not name:
+            raise ValueError("Name must not be empty!")
```

Comment:
This should be under an `if name and name == ''`? Since you can skip it for update

Reply:
ah, right